pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -139,11 +156,91 @@ func (s *NodeController) SyncCloud() error {
 			glog.Errorf("Delete node error: %s", nodeID)
 		}
 	}
+
 	return nil
 }
 
-// cloudNodes constructs and returns api.NodeList from cloudprovider.
-func (s *NodeController) cloudNodes() (*api.NodeList, error) {
+// SyncNodeStatus synchronizes cluster nodes status to master server.
+func (s *NodeController) SyncNodeStatus() error {
+	nodes, err := s.kubeClient.Nodes().List()
+	if err != nil {
+		return err
+	}
+	nodes = s.DoChecks(nodes)
+	for _, node := range nodes.Items {
+		// TODO: Skip update if nothing changes.
+		glog.Errorf("updating node %+v", node)
+		_, err = s.kubeClient.Nodes().Update(&node)
+		if err != nil {
+			glog.Errorf("error updating node %s: %v", node.Name, err)
+		}
+	}
+	return nil
+}
+
+// DoChecks performs health checking for given list of nodes.
+func (s *NodeController) DoChecks(nodes *api.NodeList) *api.NodeList {
+	for i := range nodes.Items {
+		// TODO: Add concurrency.
```

Comment:
Regarding to checks taking longer than invocation period.  I think we are fine cause util.Forever() invokes checks every period time, the time does not include the method running time.